Modules/mmapmodule.c

```diff
@@ -1117,6 +1117,46 @@ mmap_mmap_seek_impl(mmap_object *self, Py_ssize_t dist, int how)
     return NULL;
 }
 
+/*clinic*/
+
+/*[clinic input]
+mmap.mmap.set_name
+
+    name: str
+    /
+
+[clinic start generated code]*/
+
+static PyObject *
+mmap_mmap_set_name_impl(mmap_object *self, const char *name)
+/*[clinic end generated code: output=1edaf4fd51277760 input=6c7dd91cad205f07]*/
+{
+    const char* prefix = "cpython:mmap:";
+    if (strlen(name) + strlen(prefix) > 80) {
+        PyErr_SetString(PyExc_ValueError, "name is too long");
+        return NULL;
+    }
+#ifdef MAP_ANONYMOUS
+    if (self->flags & MAP_ANONYMOUS) {
+        char buf[80] = {0, };
```

Comment:
You should add +1 for the NUL character, no?

```suggestion
        char buf[81];
```